schema/query_test.go

```diff
@@ -102,6 +111,10 @@ func TestParseQuery(t *testing.T) {
 	q, err = ParseQuery("{\"foo\": {\"$nin\": [\"bar\", \"baz\"]}}", s)
 	assert.NoError(t, err)
 	assert.Equal(t, Query{NotIn{Field: "foo", Values: []Value{"bar", "baz"}}}, q)
+	q, err = ParseQuery("{\"foo\": {\"$gte\": \"2009-05-20T10:23:34+00:00\"}}", s)
```

Comment:
We need a test that test against a field with a `Time` validator. Here your test is against a non-typed field which should not allow queries against time.